addons/survey/models/survey_user_input.py

```diff
@@ -243,6 +244,13 @@ def _mark_done(self):
         Challenge = self.env['gamification.challenge'].sudo()
         badge_ids = []
         for user_input in self:
+            user_input.message_post(
+                author_id=self.env.ref('base.partner_root').id,
```

Comment:
It seems like `author_id` is constant and that the `ref` function could potentially trigger SQL queries. Even if there is a cache mechanism, I would put that call outside of the for loop and store its result in an intermediate variable that I will reuse in the for loop.

Like so:

```python
badge_ids = []
author_id = self.env.ref('base.partner_root').id
for user_input in self:
    user_input.message_post(
        author_id=author_id,
        body=_('%(participant)s just participated in %(survey_title)s.',
```

I tried understand how the `ref` function is implemented and I see that there is a cache mechanism for `_xmlid_lookup` (see the `@tools.ormcache` decorator) but I don't know if there is such mechanism for the `browse` function :thinking: